### PR TITLE
plugins/smart-splits: init

### LINKED ISSUE #1050: [PLUGIN REQUEST] smart-splits
| Field    | Description |
|----------|-------------|
| Plugin   | `smart-splits`  | 
| Homepage | `https://github.com/mrjones2014/smart-splits.nvim`     | 
| Nixpkgs  | `vimPlugins.smart-splits-nvim` |

### PR DESCRIPTION
**Pull request by foo-dogsquared**
First stab at making a (formal) NixVim plugin unlike what I'm doing at my own NixVim config. I'm guessing there is a migration at Nix RFC42-style settings so I did that.

Resolves #1050. (Probably a follow-up PR for [legendary.nvim](https://github.com/mrjones2014/legendary.nvim) as it has integration for it. Also, it looks interesting.)

Also, I cannot test this in my local setup as it always crashes so I hope you don't mind if I fully rely on the CI for the testing.

### CODE REVIEW

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Hello, this looks pretty good.
You can actually simplify the implementation using our brand new `mkNeovimPlugin` helper.
Have a look at [this PR](https://github.com/nix-community/nixvim/pull/971/files#diff-cb75710d2ff4866b9b3c544b2d823265c7e091a86e5db224aa79ac1cd62b0099) for inspiration.

**Comment by foo-dogsquared** [comment]
Converted into `mkNeovimPlugin` and also rebased it on top of f755eeb961a4d13ec72921cb4dbc4f2d6c8a58bb.

File: plugins/utils/smart-splits.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  lib,
+  pkgs,
+  helpers,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin {
+  name = "smart-splits";
+  maintainers = with lib.maintainers; [foo-dogsquared];
```
Comment (GaetanLepage) [inline plugins/utils/smart-splits.nix:9]:
```suggestion
  name = "smart-splits";
  originalName = "spart-splits.nvim";

  maintainers = with lib.maintainers; [foo-dogsquared];
```

Comment (GaetanLepage) [inline plugins/utils/smart-splits.nix:9]:
```suggestion
  name = "smart-splits";
  maintainers = [lib.maintainers.foo-dogsquared];
```

File: plugins/utils/smart-splits.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  defaultPackage = pkgs.vimPlugins.smart-splits-nvim;
+  settingsExample = {
+    resize_mode = {
+      quit_key = "<ESC>";
+      resize_keys = ["h" "j" "k" "l"];
+      silent = true;
+    };
+    ignored_events = ["BufEnter" "WinEnter"];
+  };
+}
```
Comment (GaetanLepage) [inline plugins/utils/smart-splits.nix:19]:
```suggestion
  defaultPackage = pkgs.vimPlugins.smart-splits-nvim;

  settingsExample = {
    resize_mode = {
      quit_key = "<ESC>";
      resize_keys = ["h" "j" "k" "l"];
      silent = true;
    };
    ignored_events = ["BufEnter" "WinEnter"];
  };
}
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Few nit picks but otherwise looks good !

File: plugins/utils/smart-splits.nix (hunk 1 above)
Comment (foo-dogsquared) [reply plugins/utils/smart-splits.nix:9]: Done.

File: plugins/utils/smart-splits.nix (hunk 2 above)
Comment (foo-dogsquared) [reply plugins/utils/smart-splits.nix]: Done and done.

**Comment by foo-dogsquared** [comment]
Also fixed the error in CI. Apparently, I'm not using the helper right.

**Pull request merged**